azurerm/internal/provider/provider.go

```diff
@@ -212,14 +212,6 @@ func azureProvider(supportLegacyTestSuite bool) terraform.ResourceProvider {
 
 			"features": schemaFeatures(supportLegacyTestSuite),
 
-			// Advanced feature flags
-			"skip_credentials_validation": {
-				Type:        schema.TypeBool,
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("ARM_SKIP_CREDENTIALS_VALIDATION", false),
-				Description: "This will cause the AzureRM Provider to skip verifying the credentials being used are valid.",
-			},
```

Comment:
this'd be a breaking change rather than a deprecation (although ignoring it's value is otherwise fine) - but we can deprecate this by using a feature toggle below and adding a "deprecation" message:

```
if !features.ThreePointOh() {
		p.Schema["skip_credentials_validation"] = &schema.Schema{
			Type:        schema.TypeBool,
			Optional:    true,
			DefaultFunc: schema.EnvDefaultFunc("ARM_SKIP_CREDENTIALS_VALIDATION", false),
			Description: "[DEPRECATED] This will cause the AzureRM Provider to skip verifying the credentials being used are valid.",
			Deprecated:  "This field is deprecated and will be removed in version 3.0 of the Azure Provider",
		}
	}
```